cmd/client-s3.go

```diff
@@ -863,10 +874,24 @@ func (c *s3Client) SetAccess(bucketPolicy string) *probe.Error {
 	if bucket == "" {
 		return probe.NewError(BucketNameEmpty{})
 	}
-	e := c.api.SetBucketPolicy(bucket, object, policy.BucketPolicy(bucketPolicy))
+	policyStr, e := c.api.GetBucketPolicy(bucket)
 	if e != nil {
 		return probe.NewError(e)
 	}
+	var p = policy.BucketAccessPolicy{Version: "2012-10-17"}
+	if policyStr != "" {
+		if e = json.Unmarshal([]byte(policyStr), &p); e != nil {
+			return probe.NewError(e)
+		}
+	}
+	p.Statements = policy.SetPolicy(p.Statements, policy.BucketPolicy(bucketPolicy), bucket, object)
+	policyB, e := json.Marshal(p)
+	if e != nil {
+		return probe.NewError(e)
+	}
+	if e = c.api.SetBucketPolicy(bucket, string(policyB)); e != nil {
```

Comment:
One problem here when setting 'none' policy with mc.

It can be easily reproduced:
```
mc policy download s3/vadmeste/
mc policy none s3/vadmeste
```

Reply:
FIxed